libstorage/CachedStorage.cpp

```diff
@@ -129,26 +132,29 @@ Entries::Ptr CachedStorage::select(
 
     auto entries = selectNoCondition(hash, num, tableInfo, key, condition)->entries();
 
-    for (size_t i = 0; i < entries->size(); ++i)
+    if (entries)
     {
-        auto entry = entries->get(i);
-        if (condition)
+        for (size_t i = 0; i < entries->size(); ++i)
         {
-            if (condition->process(entry))
+            auto entry = entries->get(i);
+            if (condition)
+            {
+                if (condition->process(entry))
+                {
+                    auto outEntry = std::make_shared<Entry>();
+                    outEntry->copyFrom(entry);
+
+                    out->addEntry(outEntry);
+                }
+            }
+            else
```

Comment:
for (size_t i = 0; i < entries->size(); ++i)
    {
        if (condition && !condition->process(entry))
        {
            continue;
        }
        auto outEntry = std::make_shared<Entry>();
        outEntry->copyFrom(entry);
        out->addEntry(outEntry);
    }